GWXtreme/parametrized_eos_sampler.py

```diff
@@ -179,7 +179,61 @@ def save_data(self):
         f.create_dataset('logp',data=np.array(self.logp))
         f.close()
         
+    def parse_samples(self,burn_in_frac = 0.5, thinning = None):
+        '''
+        This methods loads a .h5 file with the MCMC samples 
+        of EoS hyper-parameters and parses them for use by the 
+        plotting functions.
+        see https://emcee.readthedocs.io/en/stable/tutorials/autocorr/
+        for some documentation on choosing thinning and burn-in
+        
+        burn_in_frac  :: fraction of samples to discard from each chain
+                         for MCMC burn in. Default corresponds to discarding 
+                         half the samples in each chain.
+        
+        thinning      :: Number of samples to skip in each chain post
+                         burn in. "None" implements the default value 
+                         which is either (length of chain)/50 or half of
+                         the maximum integrated autocorrelation time. The 
+                         former is used in case autocorrelation analysis 
+                         yields throws non-convergence error. For no thinning 
+                         set thinning=1
+        
+        '''
+        
+        Ns=self.samples.shape
+        burn_in=int(Ns[0]*burn_in_frac)
+        samples=[]
+        
+        if thinning is None:
+            thinning=int(Ns[0]/50.)
 
+            try:
+                thinning=int(max(mc.autocorr.integrated_time(self.samples))/2.)
```

Comment:
How do you get access to the object `mc` in this line? It is not passed as an argument, neither is a class object.

Reply:
emcee is imported as mc on [line 24](https://github.com/AnaryaRay1/GWXtreme/blob/2dcf2303d0a031898412b91f014812551efcd1c6/GWXtreme/parametrized_eos_sampler.py#L24)